expression/builtin_compare_vec.go

```diff
@@ -707,11 +706,70 @@ func (b *builtinGreatestRealSig) vecEvalReal(input *chunk.Chunk, result *chunk.C
 }
 
 func (b *builtinLeastTimeSig) vectorized() bool {
-	return false
+	return true
 }
 
 func (b *builtinLeastTimeSig) vecEvalString(input *chunk.Chunk, result *chunk.Column) error {
-	return errors.Errorf("not implemented")
+	n := input.NumRows()
+	dst, err := b.bufAllocator.get(types.ETTimestamp, n)
+	if err != nil {
+		return err
+	}
+	defer b.bufAllocator.put(dst)
+
+	sc := b.ctx.GetSessionVars().StmtCtx
+	dst.ResizeTime(n, false)
+	dstTimes := dst.Times()
+	for i := 0; i < n; i++ {
+		dstTimes[i] = types.Time{
+			Time: types.MaxDatetime,
+			Type: mysql.TypeDatetime,
+			Fsp:  types.DefaultFsp,
+		}
+	}
+	var argTime types.Time
+
+	var findInvalidTime []bool = make([]bool, n)
+	var invalidValue []string
+
+	for j := 0; j < len(b.args); j++ {
+		if err := b.args[j].VecEvalString(b.ctx, input, result); err != nil {
+			return err
+		}
+		for i := 0; i < n; i++ {
+			if result.IsNull(i) || dst.IsNull(i) {
+				dst.SetNull(i, true)
+				continue
+			}
+			argTime, err = types.ParseDatetime(sc, result.GetString(i))
+			if err != nil {
+				if err = handleInvalidTimeError(b.ctx, err); err != nil {
+					return err
+				} else if !findInvalidTime[i] {
+					invalidValue = append(invalidValue, result.GetString(i))
+					findInvalidTime[i] = true
+				}
+				continue
+			}
+			if argTime.Compare(dstTimes[i]) < 0 {
+				dstTimes[i] = argTime
+			}
+		}
+	}
+	result.ReserveString(n)
+	for i := 0; i < n; i++ {
+		if findInvalidTime[i] {
+			result.AppendString(invalidValue[0])
```

Comment:
This implementation is not correct here.  The first string in `invalidValue` is not the one that `findInvalidTime` is true with the smallest `i`. For example, we have two vectors: `[valid, valid, invalidAAA], [valid, InvalidBBB, valid]`. findInvalidTime is:
`[false, true, true]`, but the invalidValue is: `[invalidAAA, invalidBBB]`.